config/configsource/configsource.go

```diff
@@ -0,0 +1,267 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configsource
```

Comment:
Everything that is now in `configsource` can be in an internal package that leaves in an internal directory `config/configsource/internal/`

Reply:
Instead of internal, can we name it something like `experimental`? That would allow us to implement experimental config sources in contrib. I know "experimental" does not hold any special meaning and does not provent others from using it but perhaps the word in import path should be enough to dissuade anyone from using it unless they are contributing to the same experiment. 

We could also implement initial couple of config sources in core and move them to contrib if/when they graduate but then we'll be adding new dependencies to core and not sure if we should run experiments like these in core beyond what is necessary (types of new components). //cc @pjanotti